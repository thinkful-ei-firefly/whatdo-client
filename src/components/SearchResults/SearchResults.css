.SearchResults{
    font-family: sans-serif;
    margin: 10px;
    border-radius: 7px;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    align-content: center;
    border-radius: 7px;
    /* height: 100%; */
    align-content: stretch;
    justify-content: space-around;
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
    -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    transition: all 0.4s ease 0s;
    padding: 10px;
}
.SearchResults h2{
    width: 95%;
    
    padding: 10px;
    text-align: center;
    margin:5px auto;
    border-radius: 7px ;
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
    -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    transition: all 0.4s ease 0s;
    background: rgba(250, 248, 241, 0.666);

}

.SearchResults ul {
    padding: 0px;
        /* display: inline-block;
    margin: 5px; */
/* background-color: red;     */
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    align-content: center;
    border-radius: 7px;
    /* height: 100%; */
    /* align-content: stretch; */
    justify-content: space-evenly;
    width: 95%;
    margin: auto;
}

.EventItem{
    /* display: inline-block;
    margin: 5px; */
/* background-color: red;     */
/* display: flex;
flex-direction: column;
flex-wrap: wrap;
align-content: center;
border-radius: 7px;
justify-content: space-evenly;
width: 95%;
margin: auto; */
}

/* .EventItem:hover {
    transform: scale(1.1);
    background: linear-gradient(to top, white ,rgba(82, 82, 248, 0.571));
    border-radius: 7px;
} */

/* .EventItem__Container {
    width: 95%;
    text-align: center;
    margin-left:auto;
    margin-right:auto;
    align-items: center;
    position: relative;
    min-height: 200px;
} */

.EventItem{
    position: relative;
    display: inline-block;
    padding: 15px 30px;
    /* color: #2196f3; */
    /* text-transform: uppercase; */
    /* letter-spacing: 4px; */
    text-decoration: none;
    /* font-size: 24px; */
    overflow: hidden;
    transition: 0.2s;
    width: 100%;
    border-radius: 7px;
    background-color: rgba(255, 255, 255, 0);
  }
  .EventItem:hover{
    color: #255784;
    /* background: #2196f3; */
    box-shadow: 0 0 10px #2196f3, 0 0 40px #2196f3;
    transition-delay: 1s;
    border-radius: 7px;
    transform: scale(1.05);
  }
  .EventItem span{
    position: absolute;
    display: block;
  }
  .EventItem span:nth-child(1)  {
    top: 0;
    left: -100%;
    width: 100%;
    height: 3px;
    background: linear-gradient(90deg,transparent,red);
    box-shadow: 0 0 10px #2196f3;
  }
  .EventItem:hover span:nth-child(1) {
    left: 100%;
    transition: 1s;
  }
  .EventItem span:nth-child(3) {
    bottom: 0;
    right: -100%;
    width: 100%;
    height: 3px;
    background: linear-gradient(270deg,transparent,red);
    box-shadow: 0 0 10px #2196f3 ;
  }
  .EventItem:hover span:nth-child(3) {
    right: 100%;
    transition: 1s;
    transition-delay: 0.5s;
  }
  .EventItem span:nth-child(2) {
    top: -100%;
    right: 0;
    width: 3px;
    height: 100%;
    background: linear-gradient(180deg,transparent,red);
    box-shadow: 0 0 10px #2196f3 ;
  }
  .EventItem:hover span:nth-child(2) {
    top: 100%;
    transition: 1s;
    transition-delay: 0.25s;
  }
  .EventItem span:nth-child(4) {
    bottom: -100%;
    left: 0;
    width: 3px;
    height: 100%;
    background: linear-gradient(360deg,transparent,red);
    box-shadow: 0 0 10px #2196f3 ;
  }
  .EventItem:hover span:nth-child(4) {
    bottom: 100%;
    transition: 1s;
    transition-delay: 0.75s;
  }




.fa-heart {
    color:red;
}

