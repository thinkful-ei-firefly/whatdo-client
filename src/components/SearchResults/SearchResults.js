import React from 'react'
import EventItem from '../EventItem/EventItem'
import SearchContext from '../../contexts/SearchContext'
import PageBar from '../../components/PageBar/PageBar'
import './SearchResults.css'

export default class SearchResults extends React.Component {

  static contextType = SearchContext

  render () {

    const events = this.context.events;

    if (events.length > 0) {
      const eventList = events.map(event => 
        <li key={event.id}>
          <EventItem 
            key={event.id}
            fetch_id={event.id}
            id={event.id}
            name={event.name}
            url={event.url}
            description={event.description || 'Visit the site for more information!'}
            venue={event.venue}
            address={event.address}
            city_name={event.city_name}
            region_name={event.region_name}
            start_time={event.start_time}
            stop_time={event.stop_time || event.start_time}
            image={event.image}
            saveEvent={this.context.saveEvent}
          />
        </li>
      )

      return (
        <div className="SearchResults">
        <h2>Search results for (options) within (x) miles of (location)</h2>
          <ul>
            {eventList}
          </ul>
          <PageBar />
        </div>
      )
    }
    else {
      return (
        <div className="SearchResults">
          <h2 >Sorry, we couldn't find any events near {this.context.zipCode}</h2>          
        </div>
      )
    }
    
  }
}