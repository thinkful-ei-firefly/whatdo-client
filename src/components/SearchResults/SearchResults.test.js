import React from 'react'
import ReactDOM from 'react-dom'
import SearchResults from './SearchResults'
import renderer from 'react-test-renderer'

describe('<SearchResults />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SearchResults />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders as expected', () => {
    const tree = renderer.create(<SearchResults />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
