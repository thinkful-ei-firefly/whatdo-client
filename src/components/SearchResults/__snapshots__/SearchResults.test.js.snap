// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> renders as expected 1`] = `
<div
  className="SearchResults"
>
  <h2
    className="alert"
  >
    Sorry, we couldn't find any events near 
    30084
  </h2>
</div>
`;
